elixir/apps/web/lib/web/live/sites/new_token.ex

```diff
@@ -4,12 +4,20 @@ defmodule Web.Sites.NewToken do
 
   def mount(%{"id" => id}, _session, socket) do
     with {:ok, group} <- Gateways.fetch_group_by_id(id, socket.assigns.subject) do
-      {:ok, group} =
-        Gateways.update_group(%{group | tokens: []}, %{tokens: [%{}]}, socket.assigns.subject)
+      {group, env} =
+        if connected?(socket) do
+          {:ok, group} =
+            Gateways.update_group(%{group | tokens: []}, %{tokens: [%{}]}, socket.assigns.subject)
```

Comment:
@bmanifold `mount/3` is called twice: when a static page is generated and then once more when web socket is connected. So behind the scenes if you hit refresh we will insert the token two times. This trick allows to check which kind of render this is and avoid inserting a record before WS is connected. Downside is that container will be empty for a brief moment while it loads.